kube/src/api/resource.rs

```diff
@@ -583,65 +389,67 @@ mod test {
 
     #[test]
     fn list_path() {
-        let r = Resource::namespaced::<appsv1::Deployment>("ns");
+        let url = appsv1::Deployment::url_path(&(), Some("ns"));
         let gp = ListParams::default();
-        let req = r.list(&gp).unwrap();
+        let req = Request::new(url).list(&gp).unwrap();
         assert_eq!(req.uri(), "/apis/apps/v1/namespaces/ns/deployments");
     }
     #[test]
     fn watch_path() {
-        let r = Resource::namespaced::<corev1::Pod>("ns");
+        let url = corev1::Pod::url_path(&(), Some("ns"));
         let gp = ListParams::default();
-        let req = r.watch(&gp, "0").unwrap();
+        let req = Request::new(url).watch(&gp, "0").unwrap();
         assert_eq!(
             req.uri(),
             "/api/v1/namespaces/ns/pods?&watch=true&resourceVersion=0&timeoutSeconds=290&allowWatchBookmarks=true"
         );
     }
     #[test]
     fn replace_path() {
-        let r = Resource::all::<appsv1::DaemonSet>();
+        let url = appsv1::DaemonSet::url_path(&(), None);
         let pp = PostParams {
             dry_run: true,
             ..Default::default()
         };
-        let req = r.replace("myds", &pp, vec![]).unwrap();
+        let req = Request::new(url).replace("myds", &pp, vec![]).unwrap();
         assert_eq!(req.uri(), "/apis/apps/v1/daemonsets/myds?&dryRun=All");
     }
 
     #[test]
     fn delete_path() {
-        let r = Resource::namespaced::<appsv1::ReplicaSet>("ns");
+        let url = appsv1::ReplicaSet::url_path(&(), Some("ns"));
         let dp = DeleteParams::default();
-        let req = r.delete("myrs", &dp).unwrap();
+        let req = Request::new(url).delete("myrs", &dp).unwrap();
         assert_eq!(req.uri(), "/apis/apps/v1/namespaces/ns/replicasets/myrs");
         assert_eq!(req.method(), "DELETE")
     }
 
     #[test]
     fn delete_collection_path() {
-        let r = Resource::namespaced::<appsv1::ReplicaSet>("ns");
+        let url = appsv1::ReplicaSet::url_path(&(), Some("ns"));
         let lp = ListParams::default();
         let dp = DeleteParams::default();
-        let req = r.delete_collection(&dp, &lp).unwrap();
+        let req = Request::new(url).delete_collection(&dp, &lp).unwrap();
         assert_eq!(req.uri(), "/apis/apps/v1/namespaces/ns/replicasets");
         assert_eq!(req.method(), "DELETE")
     }
 
     #[test]
     fn namespace_path() {
-        let r = Resource::all::<corev1::Namespace>();
+        let url = corev1::Namespace::url_path(&(), None);
         let gp = ListParams::default();
-        let req = r.list(&gp).unwrap();
+        let req = Request::new(url).list(&gp).unwrap();
```

Comment:
as you can see, it's slightly more error prone to use `Request` than it was to use old `Resource` due to having to match-up the urls yourself, but it is a lower level interface.